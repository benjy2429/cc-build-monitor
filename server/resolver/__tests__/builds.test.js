import resolveBuilds from '../builds';
import buildMock from '../../../mocks/builds/normal';

describe('Resolve builds', () => {
  it('parses and decorates json data', async () => {
    const mockFetch = () => Promise.resolve(buildMock);
    const result = await resolveBuilds(mockFetch);
    expect(result).toMatchSnapshot();
  });

  it('returns an empty array when no data is fetched', async () => {
    const mockFetch = () => Promise.resolve([]);
    const result = await resolveBuilds(mockFetch);
    expect(result).toMatchSnapshot();
  });

  it('returns an empty array when the fetch fails', async () => {
    const mockFetch = () => Promise.reject();
    const result = await resolveBuilds(mockFetch);
    expect(result).toMatchSnapshot();
  });
});
