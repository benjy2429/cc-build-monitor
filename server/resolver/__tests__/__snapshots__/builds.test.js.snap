// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Resolve builds parses and decorates json data 1`] = `
Array [
  Object {
    "branch": "master",
    "buildNumber": 2,
    "buildTime": 10000,
    "commit": Object {
      "hash": "abc123",
      "message": "Commit message",
    },
    "repo": "my-project",
    "status": "success",
    "user": Object {
      "avatar": "https://avatar.url",
      "name": "Me",
    },
  },
]
`;

exports[`Resolve builds returns an empty array when no data is fetched 1`] = `Array []`;

exports[`Resolve builds returns an empty array when the fetch fails 1`] = `Array []`;
